src/parse/mod.rs

```diff
@@ -150,6 +143,44 @@ impl<I: Lexer> Parser<I> {
                     data: Token::Whitespace(_),
                     ..
                 })) => continue,
+                Some(Ok(Locatable {
+                    data: Token::Literal(Literal::Str(mut concat)),
+                    mut location,
+                })) => {
+                    // 5.1.1.2p1: Translation phase 6: Adjacent string literal tokens are concatenated.
+                    loop {
+                        match self.tokens.peek() {
+                            Some(Ok(Locatable {
+                                data: Token::Literal(Literal::Str(_)),
+                                location: next_location,
+                            })) => {
+                                location = location.merge(next_location);
+                                let mut s = match self.tokens.next().unwrap().unwrap().data {
+                                    Token::Literal(Literal::Str(s)) => s,
+                                    _ => unreachable!(),
+                                };
```

Comment:
Once `Token` is Copy, I can remove this in favor of just dereferencing the borrowed token.